### PR TITLE
Convert jsonDynamic to contentitem extension

### LINKED ISSUE #16233: razor view does  allow dynamic casting of property values using system.text.json
Using the new system.text.json seems to not cast to dynamic in razor views , so you cannot access dynamically property values like previously .

See in below image when looping in a razor view over some content items i cannot extract dynamically the ContentItemId Property value like before.  is there a way to cast to dynamic like before ?

image

image

### PR DESCRIPTION
**Pull request by giannik**
resolves #16233

### CODE REVIEW

**Comment by giannik** [comment]
@gvkries please review.

**Comment by gvkries** [comment]
I think it is beneficial to add support for a generic `ToObject<T>` functionality back to `JsonDynamicObject` and probably `JsonDynamicValue` as well. This was possible with Newtonsoft and would avoid having a specialized extension method just for `ContentItem`. It would also solve problems like #16221.

It would look like in this example:
```csharp
@foreach (var foo in Model.ContentItem.Content.NamedBagPart.ContentItems)
{
    var contentItem = foo.ToObject<ContentItem>();
    ...
}
```
Note that this is actually possible right now, but only when using the `JsonObject` directly:
```csharp
@foreach (var foo in Model.ContentItem.Content.NamedBagPart.ContentItems)
{
    var contentItem = ((JsonObject)foo).ToObject<ContentItem>();
    ...
}
```
 We also have references to this style in the documentation, so we must fix it anyway IMHO.
https://github.com/OrchardCMS/OrchardCore/blob/7196b5b1c6c0180b17bb4dabe944232001b0a681/src/docs/reference/modules/Media/README.md?plain=1#L456

It would just require something like `public T? ToObject<T>() => _jsonObject.ToObject<T>();` in `JsonDynamicObject` and `JsonDynamicValue`, as there is an existing extension method for `JsonNode`.

**Comment by gvkries** [comment]
I guess `JsonDynamicArray` must be considered as well.

**Comment by giannik** [comment]
@gvkries feel free to make any changes.

File: src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentItemExtensions.cs [hunk 1]
```diff
@@ -230,5 +231,12 @@ public static ContentItem Merge(this ContentItem contentItem, object properties,
 
             return contentItem;
         }
+        public static ContentItem ToContentItem(this JsonDynamicObject jsonDynamic)
+        {
+            var contentItem = new ContentItem();
+            var contentProperties = jsonDynamic.ToString();
+            contentItem.Merge(JObject.Parse(contentProperties));
+            return contentItem;
+        }
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentItemExtensions.cs:240]:
```suggestion

        public static ContentItem ToContentItem(this JsonDynamicObject jsonDynamic)
        {
            var contentItem = new ContentItem();
            var contentProperties = jsonDynamic.ToString();
            contentItem.Merge(JObject.Parse(contentProperties));

            return contentItem;
        }
```

**Review by hishamco — CHANGES_REQUESTED** [review]

**Comment by gvkries** [comment]
> @gvkries feel free to make any changes.

I don't think I have access to your PR :) I will open another PR with a different approach.

**Comment by gvkries** [comment]
@giannik Please have a look at #16292...

**Comment by giannik** [comment]
closing in favor of  #16292

**Pull request closed without merging**